raiden/network/proxies/token_network.py

```diff
@@ -809,96 +864,241 @@ def _set_total_deposit(
+                )
+
+                channel_state = self._get_channel_state(
+                    participant1=self.node_address,
+                    participant2=partner,
+                    block_identifier=failed_at_blockhash,
+                    channel_identifier=channel_identifier,
+                )
+                token_network_deposit_limit = self.token_network_deposit_limit(
+                    block_identifier=failed_at_blockhash
+                )
+                channel_participant_deposit_limit = self.channel_participant_deposit_limit(
+                    block_identifier=failed_at_blockhash
+                )
+
+                total_channel_deposit = total_deposit + partner_details.deposit
+
+                network_balance = self.token.balance_of(
+                    Address(self.address), given_block_identifier
+                )
+
+                # Deposit was prohibited because the channel is settled
+                if channel_state == ChannelState.SETTLED:
+                    msg = "Deposit is not possible due to channel being settled"
+                    raise RaidenRecoverableError(msg)
+                # Deposit was prohibited because the channel is closed
+                elif channel_state == ChannelState.CLOSED:
+                    msg = "Channel is already closed"
+                    raise RaidenRecoverableError(msg)
+                elif channel_state == ChannelState.REMOVED:
+                    msg = "Channel was settled and unlocked"
+                    raise RaidenRecoverableError(msg)
+                elif our_details.deposit >= total_deposit:
+                    msg = "Attempted deposit has already been done"
+                    raise RaidenRecoverableError(msg)
+                # Check if deposit is being made on a nonexistent channel
+                elif channel_state == ChannelState.NONEXISTENT:
+                    msg = (
+                        f"Channel between participant {to_checksum_address(self.node_address)} "
+                        f"and {to_checksum_address(partner)} does not exist"
+                    )
+                    raise RaidenUnrecoverableError(msg)
+                elif total_channel_deposit >= UINT256_MAX:
+                    raise RaidenUnrecoverableError("Deposit overflow")
+
+                elif total_deposit > channel_participant_deposit_limit:
+                    msg = (
+                        f"Deposit of {total_deposit} exceeded the "
+                        f"channel participant deposit limit"
+                    )
+                    raise RaidenRecoverableError(msg)
+                elif network_balance + amount_to_deposit > token_network_deposit_limit:
+                    msg = (
+                        f"Deposit of {amount_to_deposit} exceeded "
+                        f"the token network deposit limit."
+                    )
+                    raise RaidenRecoverableError(msg)
+
+                raise RaidenUnrecoverableError(
```

Comment:
Here too maybe it is safer to have a recoverable error, since we call external code from our code